### PR TITLE
Enable building universal (x64,arm64) binaries for MacOS

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
Created cmakefile specifying both x64 and arm64 architectures.  Tested locally on M1 MacBookPro using `file` to verify the resulting dylib is universal.

```console
PS> file /Users/steve/repos/PowerShell-Native/src/powershell-unix/libpsl-native.dylib
/Users/steve/repos/PowerShell-Native/src/powershell-unix/libpsl-native.dylib: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit dynamically linked shared library x86_64] [arm64:Mach-O 64-bit dynamically linked shared library arm64]
/Users/steve/repos/PowerShell-Native/src/powershell-unix/libpsl-native.dylib (for architecture x86_64):	Mach-O 64-bit dynamically linked shared library x86_64
/Users/steve/repos/PowerShell-Native/src/powershell-unix/libpsl-native.dylib (for architecture arm64):	Mach-O 64-bit dynamically linked shared library arm64
```

Needed to switch to vmimage for MacOS 10.15+ which has XCode12 to support building universal binary for M1 and x64.

### CODE REVIEW

File: .vsts-ci/mac.yml [hunk 1]
```diff
@@ -9,11 +9,13 @@ variables:
...
   - repo: self
     clean: true
-phases:
-- phase: Build
 
-  queue:
-    name: Hosted macOS
+jobs:
+- job: Build
+
```
Comment (PaulHigin) [inline .vsts-ci/mac.yml:15]: Add `displayName:` for job?

**Review by PaulHigin — APPROVED** [review]
LGTM

**Review by TravisEz13 — APPROVED** [review]

File: .vsts-ci/mac.yml (hunk 1 above)
Comment (SteveL-MSFT) [reply .vsts-ci/mac.yml:15]: I'm following the convention of the other ymls which has displayname for steps, but not the job

**Comment by adityapatwardhan** [comment]
@SteveL-MSFT - can you verify that the release build succeeds? 

https://dev.azure.com/mscodehub/PowerShellCore/_build?definitionId=1099

**Comment by SteveL-MSFT** [comment]
@adityapatwardhan verified release build succeeds and library is universal

**Pull request merged**